diff --git a/src/dev/build/build_distributables.ts b/src/dev/build/build_distributables.ts
index 9ed3ee61..fda4d376 100644
--- a/src/dev/build/build_distributables.ts
+++ b/src/dev/build/build_distributables.ts
@@ -53,8 +53,6 @@ export async function buildDistributables(log: ToolingLog, options: BuildOptions
    */
   await run(Tasks.VerifyEnv);
   await run(Tasks.Clean);
-  await run(options.downloadFreshNode ? Tasks.DownloadNodeBuilds : Tasks.VerifyExistingNodeBuilds);
-  await run(Tasks.ExtractNodeBuilds);
 
   /**
    * run platform-generic build tasks
@@ -69,7 +67,6 @@ export async function buildDistributables(log: ToolingLog, options: BuildOptions
   await run(Tasks.InstallDependencies);
   await run(Tasks.RemoveWorkspaces);
   await run(Tasks.CleanPackages);
-  await run(Tasks.CreateNoticeFile);
   await run(Tasks.UpdateLicenseFile);
   await run(Tasks.RemovePackageJsonDeps);
   await run(Tasks.TranspileScss);
@@ -84,9 +81,6 @@ export async function buildDistributables(log: ToolingLog, options: BuildOptions
    * copy generic build outputs into platform-specific build
    * directories and perform platform/architecture-specific steps
    */
-  await run(Tasks.CreateArchivesSources);
-  await run(Tasks.PatchNativeModules);
-  await run(Tasks.InstallChromium);
   await run(Tasks.CleanExtraBinScripts);
   await run(Tasks.CleanNodeBuilds);
 
-- 
2.20.1

